src/App.tsx

```diff
@@ -1,14 +1,56 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterContext, SelectedTodoContext, Status } from './Store';
 
 export const App: React.FC = () => {
+  const [loadingTodos, setLoadingTodos] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const { selectedTodo } = useContext(SelectedTodoContext);
+  const { filter } = useContext(FilterContext);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoadingTodos(false));
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    let result = todos;
```

Comment:
you need to make a copy of the todos because in this case we are linked to the original todos list
```suggestion
    let result = [...todos];
```